packages/dev/inspector-v2/src/components/properties/metadataProperties.tsx

```diff
@@ -0,0 +1,287 @@
+
+            if (!this._editedMetadata) {
+                this._editedMetadata = "{}";
+            }
+
+            const parsedJson = JSON.parse(this._editedMetadata);
+            if (parsedJson) {
+                if (Object.prototype.hasOwnProperty.call(parsedJson, "gltf")) {
+                    if (!Object.prototype.hasOwnProperty.call(parsedJson.gltf, "extras")) {
+                        parsedJson.gltf.extras = {};
+                        changed = true;
+                    }
+                } else {
+                    parsedJson.gltf = { extras: {} };
+                    changed = true;
+                }
+            }
+
+            if (changed) {
+                this._editedMetadata = JSON.stringify(parsedJson, undefined, this.prettyJSON ? PrettyJSONIndent : undefined);
+                this.settingsChangedObservable.notifyObservers(this);
+            }
+        } catch (error) {}
+    }
+
+    save() {
+        if (this._editedMetadata) {
+            if (this.isParsable) {
+                const parsed = JSON.parse(this._editedMetadata);
+                if (!IsString(parsed)) {
+                    this._setMetadata(parsed);
+                    return;
+                }
+            }
+
+            if (this.entityType === MetadataTypes.STRING) {
+                if (this._editedMetadata !== "") {
+                    this._setMetadata(this._editedMetadata);
+                    return;
+                }
+            }
+
+            // Object type or unparseable JSON. Leave as string.
+            this._setMetadata(this._editedMetadata);
+            return;
+        }
+
+        this._setMetadata(null);
+    }
+
+    private _setMetadata(value: any) {
+        if (this.entity.metadata !== value) {
+            this.entity.metadata = value;
+
+            this._editedMetadata = this.parsedMetadata;
+
+            this.settingsChangedObservable.notifyObservers(this);
+        }
+    }
+
+    private static _Instance: Nullable<MetadataUtils> = null;
+    private static _PrettyJSON = false;
+    private static _PreventObjectCorruption = true;
+
+    public static get Instance(): MetadataUtils {
+        if (!MetadataUtils._Instance) {
+            throw new Error("MetadataUtils not initialized.");
+        }
+        return MetadataUtils._Instance;
+    }
+
+    public static set Entity(entity: IMetadataContainer) {
+        if (!MetadataUtils._Instance || MetadataUtils._Instance.entity !== entity) {
+            MetadataUtils._Instance = new MetadataUtils(entity);
+        }
+    }
+}
+
+/**
+ * Component to display metadata properties of an entity.
+ * @param props - The properties for the component.
+ * @returns A React component that displays metadata properties.
+ */
+export const MetadataProperties: FunctionComponent<{ entity: IMetadataContainer }> = (props) => {
+    const { entity } = props;
+
+    MetadataUtils.Entity = entity;
+    const metadataUtils = MetadataUtils.Instance;

```

Comment:
This is using a `static` variable to make `metadataUtils` persistent across calls to this function with the same `entity`. This is done to retain the content of the edited text area.